### ISSUE TITLE
Mongoose map (of array) exceeds call stack

### ISSUE THREAD

**Issue by www-chique** [francescov1/mongoose-tsgen#63]
This is very similar to #28 .

Hey @francescov1 - I ran into the same problem with another Map.
I am using an array of Number as the value of the map.
The error is: "RangeError: Maximum call stack size exceeded".

```
new Schema(
	{
		chapters: {
			type: Map,
			of: [Number],
		},
	},
	{
		timestamps: false,
	}
);
```

Example doc:
```
{
     _id: ".....",
     chapters: {
          "chapter1": [5, 8, 10, 12],
          "chapter2": [1, 2, 20],
     }
}
```

**Comment by francescov1** [comment]
Hey @www-chique, sorry for the delay! I've got a fix mostly done but need to clean up a few things, looking to push by end of weekend hopefully!

**Comment by francescov1** [comment]
Fixed in 8.3.6 🚀
